src/Exceptions.hpp

```diff
@@ -31,6 +32,23 @@ struct TimeoutError : std::runtime_error
 };
 
 }
-
+class MockContextException : public std::exception
+{  
+    public:  
+        const char * what() const throw()  
+        {  
+            return "IODRIVERS_BASE_MOCK Error: Expectation set outside Mock Context! Please call IODRIVERS_BASE_MOCK() before setting expecations\n";  
+        }  
+};  
+
+class TestEndsWithExpectationsLeftException : public std::exception
+{
+       public:  
+        const char * what() const throw()  
+        {  
+            return "IODRIVERS_BASE_MOCK Error: Test reached its end without satisfying all expecations.\n";  
```

Comment:
Typo. expecation -> expectations

Also, don't add a newline at the end of exception messages. It's not up to you to decide whether there should be one.